apm/otel_monitor.go

```diff
@@ -136,7 +141,13 @@ func (m *monitor) Started(ctx context.Context, evt *event.CommandStartedEvent) {
 	}
 	if !m.cfg.CommandAttributeDisabled {
 		if stmt := m.cfg.CommandTransformerFunc(evt.Command); stmt != "" {
-			attrs = append(attrs, semconv.DBStatement(stmt))
+			if formattedStmt, err := extractStatement(evt.CommandName, stmt, false); err == nil && formattedStmt != "" {
+				attrs = append(attrs, semconv.DBStatement(formattedStmt))
+			}
+
+			if strippedStatement, err := extractStatement(evt.CommandName, stmt, true); err == nil && strippedStatement != "" {
```

Comment:
Could extractStatement just return both the formatted and stripped statement in one call? That way we only have to go through marshalling / unmarshalling one time. I think we could use raw.Copy here

Reply:
Restructured to only parse the string once.

Reply:
Restructured more so it's really only parsed once.